lib/routes/meituan/tech.ts

```diff
@@ -0,0 +1,87 @@
+import { Route } from '@/types';
+import { CheerioAPI, Element, load } from 'cheerio';
+import got from '@/utils/got';
+import { parseDate } from '@/utils/parse-date';
+import cache from '@/utils/cache';
+
+const DOMAIN = 'tech.meituan.com';
+
+export const route: Route = {
+    path: '/tech',
+    categories: ['programming', 'blog'],
+    example: '/meituan/tech',
+    parameters: {},
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['tech.meituan.com'],
+        },
+    ],
+    name: '美团技术团队博客',
+    url: DOMAIN,
+    maintainers: ['ktKongTong'],
+    handler,
+};
+
+const extractPostInfo = ($: CheerioAPI, postEle: Element) => {
+    const title = $(postEle).find('.post-title').text().trim();
+    const authors = $(postEle).find('.m-post-nick').text().trim();
+    const date = $(postEle)
+        .find('.m-post-date')
+        .text()
+        .replaceAll(/[年月]/g, '-')
+        .replace('日', '')
+        .trim();
+    const description = $(postEle).find('.post-content').text();
+    const tags = $(postEle).find('.tag-links').text().trim();
+    const link = $(postEle).find('a.more-link').attr('href')!;
+    return {
+        title,
+        link,
+        description: description.replace(/阅读全文$/, '').trim(),
+        pubDate: parseDate(date),
+        author: authors,
+        category: tags.split(',').map((tag) => tag.trim()),
+    };
+};
+
+const extractPostContent = async (link: string) => {
+    const { data: response } = await got(link);
+    const $ = load(response);
+    return $('.post-content').first().html()!;
+};
+
+async function handler(ctx) {
+    const baseUrl = `https://${DOMAIN}`;
+    const { data: response } = await got(baseUrl);
+    const $ = load(response);
+    const postEles = $('.post-container-wrapper div.post-container').toArray();
+    let posts = postEles.map((el) => extractPostInfo($, el));
+
+    const isFullTextMode = ctx.req.query('mode')?.toLowerCase() === 'fulltext';
```

Comment:
Don't use middleware's `mode`, you will be sending 2 requests per posts instead of 1.